relay/nakama/cardinal.go

```diff
@@ -40,3 +44,50 @@ func getCardinalEndpoints() (txEndpoints []string, queryEndpoints []string, err
 	queryEndpoints = ep.QueryEndpoints
 	return txEndpoints, queryEndpoints, err
 }
+
+func makeRequestAndReadResp(
+	ctx context.Context,
+	notifier *receipt.Notifier,
+	endpoint string,
+	payload io.Reader,
+) (res string, err error) {
+	req, err := http.NewRequestWithContext(
+		ctx,
+		http.MethodPost,
+		utils.MakeHTTPURL(endpoint, globalCardinalAddress),
+		payload,
+	)
+	req.Header.Set("Content-Type", "application/json")
+	if err != nil {
+		return res, eris.Wrapf(err, "request setup failed for endpoint %q", endpoint)
+	}
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return res, eris.Wrapf(err, "request failed for endpoint %q", endpoint)
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode != http.StatusOK {
+		body, err := io.ReadAll(resp.Body)
+		if err != nil {
+			return res, eris.Wrapf(err, "failed to read response body, bad status code: %s: %s", resp.Status, body)
+		}
+		return res, eris.Errorf("bad status code: %d: %s", resp.StatusCode, body)
+	}
+	bz, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return res, eris.Wrap(err, "can't ready body")
```

Comment:
you were more descriptive in the error returned in the first call to `ReadAll`, I prefer that error to this one